### ISSUE TITLE
don't rely on drand networks for past beacons

### ISSUE THREAD

**Issue by LesnyRumcajs** [ChainSafe/forest#4838]
# Summary

An excellent summary is provided in https://github.com/filecoin-project/lotus/issues/12527#issuecomment-2388289290

tl;dr we shouldn't rely on drand providing endpoints for historical beacons. It's both not scalable, incorrect in certain epochs and not reliable.

The downside of this is reduced performance when requesting beacon entries for really old epochs, e.g., first epoch of calibnet.

# Completion Criteria

- [ ] Identify places where we call the drand servers for beacon data - if the data is available on chain, we should not do this and get that tipset's drand entry. There is at least one - https://github.com/ChainSafe/forest/blob/5fd1ec506ee7db4d83fa8aca14338429bf11938c/src/rpc/methods/state.rs#L1725-L1758
- [ ] assert that performance is similar to Lotus's, both calibnet and mainnet.

# Additional Links & Resources

**Comment by LesnyRumcajs** [comment]
Also, on mainnet, getting old entry doesn't work.

```
❯ time curl --silent -X POST -H "Content-Type: application/json" \
             --data '{"jsonrpc":"2.0","id":2,"method":"Filecoin.StateGetBeaconEntry","params": [ 1 ] }' \
             "http://127.0.0.1:2345/rpc/v1"
{"jsonrpc":"2.0","id":2,"error":{"code":-32603,"message":"Aggregated errors:\n"}}
________________________________________________________
Executed in  884.48 secs      fish           external
   usr time    8.36 millis  384.00 micros    7.98 millis
   sys time   21.31 millis   92.00 micros   21.21 millis
```

**Comment by LesnyRumcajs** [comment]
We have a epoch to tipset mapping so it shouldn't be that slow.
